components-api/src/main/scala/pl/touk/nussknacker/engine/api/Context.scala

```diff
@@ -2,20 +2,35 @@ package pl.touk.nussknacker.engine.api
 
 import pl.touk.nussknacker.engine.api.process.ProcessName
 
+import java.util.UUID
 import scala.jdk.CollectionConverters._
 
 object Context {
 
-  def apply(id: ContextId): Context = Context(id, Map.empty, None)
+  def apply(id: ContextId): Context = Context(id, Map.empty, None, None)
 
   def apply(id: ContextId, variables: Map[String, Any]): Context =
-    Context(id, variables, None)
+    Context(id, variables, None, None)
 
   // The dummy Context should only be used in test suites, perhaps also Nu scenario test mechanism or examples
   def dummy: Context = Context(ContextId.dummy)
 
 }
 
+object TraceId {
+
+  def generateTraceId() =
```

Comment:
What about simple `generate`? it is companion object that return `TraceId` type so we can simplify that - WDYT?